ion-schema/src/types.rs

```diff
@@ -461,13 +507,29 @@ impl TypeDefinitionImpl {
                 },
             };
 
-            let isl_constraint = IslConstraintImpl::from_ion_element(
-                isl_version,
-                "type",
-                &Element::symbol(Symbol::from("any")),
-                &isl_type_name,
-                &mut vec![],
-            )?;
+            let isl_constraint: IslConstraintImpl = match isl_version {
```

Comment:
adds default type based on ISL version (`$any` for ISL 2.0 and `any` for ISL 1.0)

Reply:
> Can you explain why this is necessary? `$any` is the top type, so adding `type: $any` in Ion Schema 2.0 should be a no-op.
I did not understand this. What did you mean by adding `type: $any` should be a no op?
IMO we are adding a default type constraint when there are none present in a top level named type definition. Hence I am adding `type: $any` for ISL 2.0 here.